src/SonarScanner.MSBuild.Common/ProcessRunnerArguments.cs

```diff
@@ -216,17 +218,18 @@ private static string ShellEscape(string argLine)
             var sb = new StringBuilder();
             foreach (var c in argLine)
             {
-                // This escape is required after %* is expanded to prevent command injections
-                sb.Append('^');
-                sb.Append('^');
-
-                // This escape is required only to pass the argument line to the batch script
-                sb.Append('^');
+                if (IsSpecialCharacter(c))
+                {
+                    sb.Append('^');
+                }
                 sb.Append(c);
             }
             return sb.ToString();
         }
-
+        private static bool IsSpecialCharacter(char c)
```

Comment:
Since this method is used only in 1 place and it is pretty simple. I think you can remove it and do the check directly.

Reply:
Ok, done!